Automattic/woocommerce-payments
# Fix Tooltip rates display when there is a promo fee active

Fixes #5367

#### Changes proposed in this Pull Request

The PR does the following changes: 

* It makes the tool tip show correct `Base` and `Total per transaction` fees within `Payments > Settings > Payment Methods` section.  There was a bug that made the tool tip have the fees missing whenever there was a promotional fee applied. 
* The PR also fixes the Pill component to show custom base fees, if applicable. Else it will show the rack rate base fee ( currently 2.9% + $0.30 

#### Assumption 

If there is a promo fee applied, the discount object would contain a `discount` property, and no `percentage_rate` nor `fixed_rate` .  Hence, the PR uses the absence of these as a way to identify a promo in action vs a custom fee. Kindly let me know if this is valid. Happy to rework, otherwise. Indicating [reference here](https://github.com/Automattic/woocommerce-payments/blob/67788e6cbf069e843d709e18e20d3d849eb8a7fd/client/utils/account-fees.tsx#L251)

#### Screenshots

<details><summary>

##### Before: 

</summary>
<img width="437" alt="image" src="https://user-images.githubusercontent.com/4162931/211346645-8ce4a6e7-4599-4bb6-8d77-43b130f7793b.png">

</details>
<details><summary>

##### After: 

</summary>

<img width="437" alt="image" src="https://user-images.githubusercontent.com/4162931/211347252-a419a5c8-856f-4388-b31f-edd1972bc9b7.png">

</details>

#### Testing instructions

* Checkout to this branch and apply a promo fee to the test site - say - `wcpay-promo-2022-us-incentive-20-off`
* Browse to `Payments > Settings > Payment Methods` section
* The Pill should read `From 2.9% + $0.30` ( rack rate base fees ) 
* Mouse over the pill, and you should see the same rate as Base fee, and correctly calculated `Total per transaction fee`
* As an added check, browse to `Payments > Overview` . The `Account Details` section should show a sub-section of `Active discounts`. This section would show the discounted rate ( with rack rates struck off ), and a progress bar indicating utilization of volume limit. 
* Change the fees to a custom fee and repeat the test steps above
* This time the pill, and the tooltip mouseover, should both show the custom base fee. ( check screenshot below ) 
<img width="422" alt="image" src="https://user-images.githubusercontent.com/4162931/211350105-fccdca9f-8f01-4bc3-8edb-898048170bff.png">


-------------------


**Post merge**
